modules/product/src/main/java/com/opengamma/strata/product/SecurityInfo.java

```diff
@@ -0,0 +1,414 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ImmutableMap;
+import com.opengamma.strata.basics.market.ReferenceData;
+import com.opengamma.strata.collect.Messages;
+
+/**
+ * Additional information about a security.
+ * <p>
+ * This allows additional information about a security to be provided if desired.
+ * All properties on this class are optional.
+ */
+@BeanDefinition(builderScope = "private")
+public final class SecurityInfo
+    implements ImmutableBean, Serializable {
+
+  /**
+   * The security identifier.
+   * <p>
+   * This identifier uniquely identifies the security within the system.
+   * It is the key used to lookup the security in {@link ReferenceData}.
+   * <p>
+   * A real-world security will typically have multiple identifiers.
+   * The only restriction placed on the identifier is that it is sufficiently
+   * unique for the reference data lookup. As such, it is acceptable to use
+   * an identifier from a well-known global or vendor symbology.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final SecurityId id;
```

Comment:
It is not optional. There is a method to get the identifier from the security, but it delegates to the info object. This approach was closer to that used in `TradeInfo` and seemed to work well.